src/__init__.py

```diff
@@ -18090,18 +18082,17 @@ def get_tessdata() -> str:
         message("Tesseract-OCR is not installed")
         return False
 
-    # determine tessdata via iteration over subfolders
-    tessdata = None
-    for sub_response in response.iterdir():
-        for sub_sub in sub_response.iterdir():
-            if str(sub_sub).endswith("tessdata"):
-                tessdata = sub_sub
-                break
-    if tessdata is not None:
+    # search tessdata in folder structure
+    dirname = response[1]  # contains tesseract-ocr installation folder
+    if isinstance(dirname, bytes):  # has probably 'bytes' format
+        dirname = dirname.decode()
+    dirname = pathlib.Path(dirname)  # make a Path from it
+    sub = list(dirname.iterdir())[0]  # includes the version sub-folder
+    tessdata = str(list(sub.iterdir())[0])  # this should be the tessdata
+    if tessdata.endswith("tessdata"):  # expected name suffix
```

Comment:
This is quite difficult to understand, at least for me.

Could we instead do:
```
tessdatas = glob.glob(f'{dirname}/*/tessdata')
if len(tessdatas) == 0:
    message("unexpected: tesseract-ocr has no 'tessdata' folder")
    return False
else:
    # return latest.
    tessdatas.sort()
    return tessdata[-1]
```
?

Reply:
Made both changes.